cloud/scope/cluster.go

```diff
@@ -201,15 +232,45 @@ func (s *ClusterScope) ReserveFIP() (*vpcv1.FloatingIP, error) {
 }
 
 func (s *ClusterScope) ensureFIPUnique(fipName string) (*vpcv1.FloatingIP, error) {
-	listFloatingIpsOptions := &vpcv1.ListFloatingIpsOptions{}
-	floatingIPs, _, err := s.IBMVPCClient.ListFloatingIps(listFloatingIpsOptions)
+	var floatingIP *vpcv1.FloatingIP
+	f := func(start string) (bool, string, error) {
+		// check for existing floatingIPs
+		listFloatingIpsOptions := &vpcv1.ListFloatingIpsOptions{}
+		if start != "" {
+			listFloatingIpsOptions.Start = &start
+		}
+
+		floatingIPsList, _, err := s.IBMVPCClient.ListFloatingIps(listFloatingIpsOptions)
+		if err != nil {
+			return false, "", err
+		}
+
+		if floatingIPsList == nil {
+			return false, "", fmt.Errorf("floatingIP list returned is nil")
+		}
+
+		if len(floatingIPsList.FloatingIps) > 0 {
+			for i, floatingIPs := range floatingIPsList.FloatingIps {
+				if (*floatingIPs.Name) == fipName {
+					floatingIP = &floatingIPsList.FloatingIps[i]
+					return true, "", nil
+				}
+			}
```

Comment:
```suggestion
			for _, fp := range floatingIPsList.FloatingIps {
				if (*fp.Name) == fipName {
					floatingIP = fp
					return true, "", nil
				}
			}
```

Reply:
@mkumatag above change `floatingIP = fp` throws following lint error (should we avoid liniting on that line or try some workaround.
```
G601: Implicit memory aliasing in for loop. (gosec)
                                floatingIP = &fp
```

`var floatingIP *vpcv1.FloatingIP`

Reply:
the existing logic is using index to avoid the lint, another possible way (something like this)-
```
go func(i vpcv1.Instance) {
       instance = &i
}(ins)
```

Reply:
lets stick to indexing in that case..

Reply:
okay, will push changes.

Reply:
Fixed.